### PR TITLE
Implement BC compatibel BackedEnum switch

### PR DESCRIPTION
**Pull request by SerethiX**

### CODE REVIEW

**Comment by SerethiX** [comment]
Fixes  #10132

File: lib/Doctrine/ORM/Internal/Hydration/ObjectHydrator.php [hunk 1]
```diff
@@ -281,9 +281,13 @@ private function getEntityFromIdentityMap(string $className, array $data)
             $idHash = '';
 
             foreach ($class->identifier as $fieldName) {
-                $idHash .= ' ' . (isset($class->associationMappings[$fieldName])
-                    ? $data[$class->associationMappings[$fieldName]['joinColumns'][0]['name']]
-                    : $data[$fieldName]);
+                if (isset($class->associationMappings[$fieldName])) {
+                    $idHash .= ' ' . $data[$class->associationMappings[$fieldName]['joinColumns'][0]['name']];
+                } elseif (class_exists('\BackedEnum') && is_a($data[$fieldName], '\BackedEnum')) {
```
Comment (derrabus) [inline lib/Doctrine/ORM/Internal/Hydration/ObjectHydrator.php:286]: The `class_exists()` feels unnecessary and also wrong, given that `BackedEnum` is actually an interface.

**Review by derrabus — CHANGES_REQUESTED** [review]
Thank you, but as always: we need a test that covers your change.

**Comment by derrabus** [comment]
Closing due to lack of feedback and given that the patch is apparently not correct. Feel free to open a new PR if you want to pick up the work.

**Pull request closed without merging**